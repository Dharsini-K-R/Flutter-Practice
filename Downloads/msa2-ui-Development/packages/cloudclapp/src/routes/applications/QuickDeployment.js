import React, { useState } from "react";
import { Grid, Typography, Button, makeStyles } from "@material-ui/core";
import { useTranslation } from "react-i18next";
import { useSelector } from "react-redux";
import QuickDeploymentDialog from "cloudclapp/src/components/quick-deployment-dialog/QuickDeploymentDialog";
import {
  getConnectionSummary,
  getEnvironments,
} from "cloudclapp/src/store/designations";

const useStyles = makeStyles(({ overrides, palette }) => ({
  deploymentText: {
    boxSizing: "border-box",
    fontSize: "normal",
    color: palette.background.subTextGrey,
    textAlign: "left",
    lineHeight: "16px",
    paddingBottom: "1%",
    paddingLeft: "2%",
    paddingTop: "3%",
  },
  deploymentButton: {
    padding: "12px 20px 12px 20px",
    borderRadius: "4px",
    border: "1px solid #327bf6",
    boxSizing: "border-box",
    fontWeight: "500",
    color: overrides.MuiButton.root.color,
    textAign: "center",
    lineHeight: "normal",
    textTransform: "none",
  },
}));

const QuickDeployment = ({ defaultContext, selectedCloud, imageType }) => {
  const classes = useStyles();
  const { t } = useTranslation();
  const [showDialog, setShowDialog] = useState(false);
  const connectionSummary = useSelector(getConnectionSummary);
  const environments = useSelector(getEnvironments());

  return (
    <>
      {showDialog && (
        <QuickDeploymentDialog
          selectedCloud={selectedCloud}
          imageType={imageType}
          onClose={() => {
            setShowDialog(false);
          }}
          defaultContext={defaultContext}
          connectionSummary={connectionSummary}
          environments={environments}
        />
      )}
      <Grid
        container
        spacing={2}
        justifyContent="space-between"
        alignItems="center"
        direction="row"
      >
        <Grid item xs={3}>
          <Typography
            id="choose_application_to_deploy"
            className={classes.deploymentText}
          >
            {" "}
            {t("Choose the application you want to deploy")}{" "}
          </Typography>
        </Grid>
        <Grid item xs={3}>
          <Button
            id="dockerhub-deployment-button"
            variant="outlined"
            className={classes.deploymentButton}
            onClick={() => {
              setShowDialog(true);
            }}
          >
            {t("Create a Quick Deployment")}
          </Button>
        </Grid>
      </Grid>
    </>
  );
};

export default QuickDeployment;
