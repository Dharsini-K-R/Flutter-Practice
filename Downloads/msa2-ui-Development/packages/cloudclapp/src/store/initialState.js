import { initialState as settings } from "./settings";
import { initialState as auth } from "./auth";
import { initialState as designations } from "./designations";

export default {
  auth,
  settings,
  designations,
};
