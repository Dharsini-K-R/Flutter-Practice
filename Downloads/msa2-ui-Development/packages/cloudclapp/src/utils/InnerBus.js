import EventEmitter from "events";

// List of all events available for usage with inner bus
const EVENTS = {
  LOGOUT: "LOGOUT",
  LOGIN: "LOGIN",
};

class InnerBus extends EventEmitter {
  evt = EVENTS;
}

export default new InnerBus();
