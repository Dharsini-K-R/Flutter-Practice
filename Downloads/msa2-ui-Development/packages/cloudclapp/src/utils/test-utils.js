import React from "react";
import { render } from "@testing-library/react";
import { renderHook, act } from "@testing-library/react-hooks";
import userEvent from "@testing-library/user-event";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import { BrowserRouter as Router } from "react-router-dom";
import { StylesProvider } from "@material-ui/core";
import { ThemeProvider } from "@material-ui/core";
import SnackbarProvider from "cloudclapp/src/SnackbarProvider";
import createTheme from "cloudclapp/src/styles/theme";
import thunk from "redux-thunk";
const middlewares = [thunk];

const msaTheme = createTheme();

// Make generated classNames consistent for snapshots
const generateClassName = (rule, styleSheet) =>
  `${styleSheet.options.classNamePrefix}-${rule.key}`;

const AllTheProviders = ({ children }) => {
  return (
    <StylesProvider generateClassName={generateClassName}>
      <ThemeProvider theme={msaTheme}>
        <SnackbarProvider>{children}</SnackbarProvider>
      </ThemeProvider>
    </StylesProvider>
  );
};

const customRender = (component, options, initialRoute = "/") => {
  const componentWithRouter = (
    <Router basename={initialRoute}>{component}</Router>
  );

  return render(componentWithRouter, { wrapper: AllTheProviders, ...options });
};

const renderWithProvider = ({
  childComponent,
  mockStore,
  mockState = {},
  initialRoute,
}) => {
  const store = mockStore || configureMockStore(middlewares)(mockState);

  return {
    ...customRender(
      <Provider store={store}>{childComponent}</Provider>,
      null,
      initialRoute,
    ),
    store,
  };
};

const customRenderHook = (callback, options = {}) => {
  const wrapper = options.wrapper || AllTheProviders;

  return renderHook(callback, { ...options, wrapper });
};

// re-export everything
export * from "@testing-library/react";

// override render and renderHook methods
export {
  customRender as render,
  renderWithProvider,
  customRenderHook as renderHook,
  act as actOnHook,
  userEvent,
};
