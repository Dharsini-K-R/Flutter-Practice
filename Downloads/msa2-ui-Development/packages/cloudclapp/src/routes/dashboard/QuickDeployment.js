import React, { useState, useEffect } from "react";
import {
  Grid,
  Card,
  CardContent,
  Button,
  makeStyles,
  Typography,
  Box,
} from "@material-ui/core";
import { useSelector } from "react-redux";
import { useTranslation } from "react-i18next";
import {
  CloudIcon,
  PlugIcon,
  AngleRightIcon,
  StreamIcon,
  CloudUploadIcon,
} from "react-line-awesome";
import QuickDeploymentDialog from "../../components/quick-deployment-dialog/QuickDeploymentDialog";
import { getEnvironments } from "cloudclapp/src/store/designations";
import { getDeploymentsCount } from "../../services/Environment";
import QuickDeploymentStatusBadgeIcon from "cloudclapp/src/components/status-badge-icon/QuickDeploymentStatusBadgeIcon";
import { QUICK_DEPLOYMENT_STEPS } from "cloudclapp/src/Constants";
import isEmpty from "lodash/isEmpty";
import { getConnectionSummary } from "cloudclapp/src/store/designations";

const useStyles = makeStyles((theme) => {
  const { palette } = theme;

  return {
    button: {
      color: palette.primary.main,
      "&.MuiButton-outlined": {
        border: `1px solid ${palette.primary.main}`,
      },
    },
    flexSpaceBtw: {
      display: "flex",
      justifyContent: "space-between",
      alignItems: "center",
    },
    quickDeployCard: {
      marginTop: "20px",
      borderRadius: "10px",
      padding: "10px",
    },
    quickDepIcons: {
      width: "35px",
      height: "35px",
      color: palette.secondary.main,
      padding: "15px",
      fontSize: "xx-large",
    },
    quickDepArrows: {
      width: "10px",
      height: "18px",
      color: palette.border.main,
      padding: "15px",
      fontSize: "x-large",
    },
    collapseBtn: {
      cursor: "pointer",
    },
  };
});

const QuickDeployment = ({ cloudConnectionCount = 0 }) => {
  const classes = useStyles();
  const { t } = useTranslation();
  const [deploymentDialog, setDeploymentDialog] = useState(false);

  const [collapse, setCollapse] = useState(false);

  const connectionSummary = useSelector(getConnectionSummary);

  const environments = useSelector(getEnvironments());

  const deploymentsCount = getDeploymentsCount(environments);

  const [selectedCloud, setSelectedCloud] = useState({});

  useEffect(() => {
    setCollapse(deploymentsCount === 0);
  }, [environments, deploymentsCount]);

  useEffect(() => {
    if (connectionSummary) {
      const connectedCloud = connectionSummary.filter((cloudVendor) => {
        return cloudVendor.connections.some(
          (connection) => !isEmpty(connection.connectionName),
        );
      });
      setSelectedCloud(connectedCloud[0]);
    }
  }, [connectionSummary]);

  const [currentStep, setCurrentStep] = useState(
    QUICK_DEPLOYMENT_STEPS.CLOUD_CONNECTION_STEP,
  );

  const quickDeployExpnd = () => {
    setCollapse(!collapse);
  };

  const handleQuickDeploymentOnClick = (step) => {
    setCurrentStep(step);
    setDeploymentDialog(true);
  };

  const getEnvironmentsName = () => {
    let environmentsCreated = environments[0].envName;
    if (environments.length === 1) {
      return environmentsCreated;
    }
    environmentsCreated = environmentsCreated + ", " + environments[1].envName;
    if (environments.length > 2) {
      environmentsCreated += " ...";
    }
    return environmentsCreated;
  };
  const getCloudConnectionNames = () => {
    if (isEmpty(selectedCloud)) {
      let connectionsCreated = connectionSummary[0]?.vendorDisplayName;
      if (connectionSummary.length === 1) {
        return connectionsCreated;
      }
      connectionsCreated =
        connectionsCreated + ", " + connectionSummary[1]?.vendorDisplayName;
      if (connectionSummary.length > 2) {
        connectionsCreated += " ...";
      }
      return connectionsCreated;
    } else {
      return selectedCloud.vendorDisplayName;
    }
  };

  return (
    <Grid
      container
      spacing={3}
      direction="row"
      justifyContent="space-between"
      alignItems="stretch"
    >
      <Grid item xs={12}>
        <Card className={classes.quickDeployCard}>
          <CardContent>
            <div className={classes.flexSpaceBtw}>
              <Typography gutterBottom variant="h5">
                {t("Quick Deployment")}
              </Typography>
              <Typography
                gutterBottom
                variant="body1"
                onClick={quickDeployExpnd}
                id="DASHBOARD_COLLAPSE_BUTTON"
                className={classes.collapseBtn}
              >
                {collapse ? t("Collapse") : t("Expand")}
              </Typography>
            </div>
            {collapse ? (
              <>
                <Typography gutterBottom variant="subtitle1">
                  {t("4 Easy steps to get your application to the cloud.")}
                </Typography>
                <Grid
                  container
                  justifyContent="space-between"
                  alignItems="center"
                  style={{ paddingTop: 1 }}
                >
                  <Box
                    sx={{
                      display: "flex",
                      flexDirection: "column",
                      justifyContent: "space-between",
                      alignItems: "center",
                    }}
                  >
                    {deploymentsCount === 0 && cloudConnectionCount > 0 ? (
                      <QuickDeploymentStatusBadgeIcon
                        size="large"
                        icon={PlugIcon}
                        status={"COMPLETED"}
                        label="Cloud Connected"
                        selectedValue={getCloudConnectionNames()}
                      />
                    ) : (
                      <>
                        <PlugIcon className={classes.quickDepIcons} />
                        <Button
                          variant="outlined"
                          id="DASHBOARD_QUICK_DEPLOYMENT_BTN"
                          className={classes.button}
                          onClick={() =>
                            handleQuickDeploymentOnClick(
                              QUICK_DEPLOYMENT_STEPS.CLOUD_CONNECTION_STEP,
                            )
                          }
                        >
                          {t("Add a Cloud Connection")}
                        </Button>
                      </>
                    )}
                  </Box>
                  <AngleRightIcon className={classes.quickDepArrows} />
                  <Box
                    sx={{
                      display: "flex",
                      flexDirection: "column",
                      justifyContent: "space-between",
                      alignItems: "center",
                    }}
                  >
                    {deploymentsCount === 0 && environments.length > 0 ? (
                      <QuickDeploymentStatusBadgeIcon
                        size="large"
                        icon={CloudIcon}
                        status={"COMPLETED"}
                        label="Environment Created"
                        selectedValue={getEnvironmentsName()}
                      />
                    ) : (
                      <>
                        <CloudIcon className={classes.quickDepIcons} />
                        <Button
                          variant="outlined"
                          id="DASHBOARD_QUICK_DEPLOYMENT_ENVIRONMENT_BTN"
                          className={classes.button}
                          onClick={() =>
                            handleQuickDeploymentOnClick(
                              QUICK_DEPLOYMENT_STEPS.ENVIRONMENT_STEP,
                            )
                          }
                        >
                          {t("Add an Environment")}
                        </Button>
                      </>
                    )}
                  </Box>
                  <AngleRightIcon className={classes.quickDepArrows} />
                  <Box
                    sx={{
                      display: "flex",
                      flexDirection: "column",
                      justifyContent: "space-between",
                      alignItems: "center",
                    }}
                  >
                    <StreamIcon className={classes.quickDepIcons} />
                    <Button
                      variant="outlined"
                      id="DASHBOARD_QUICK_DEPLOYMENT_DEPLOYMENT_BTN"
                      className={classes.button}
                      onClick={() =>
                        handleQuickDeploymentOnClick(
                          environments.length === 1
                            ? QUICK_DEPLOYMENT_STEPS.DEPLOYMENT_STEP
                            : QUICK_DEPLOYMENT_STEPS.ENVIRONMENT_STEP,
                        )
                      }
                    >
                      {t("Add a Deployment")}
                    </Button>
                  </Box>
                  <AngleRightIcon className={classes.quickDepArrows} />
                  <Box
                    sx={{
                      display: "flex",
                      flexDirection: "column",
                      justifyContent: "space-between",
                      alignItems: "center",
                    }}
                  >
                    <CloudUploadIcon className={classes.quickDepIcons} />
                    <Typography gutterBottom variant="subtitle1">
                      {t("Deploy on the Cloud")}
                    </Typography>
                  </Box>
                </Grid>
              </>
            ) : (
              <div className={classes.flexSpaceBtw}>
                <Typography gutterBottom variant="subtitle1">
                  {t("Choose the Application you want to deploy")}
                </Typography>
                <Button
                  variant="outlined"
                  id="DASHBOARD_QUICK_DEPLOYMENT_BTN"
                  className={classes.button}
                  onClick={() =>
                    handleQuickDeploymentOnClick(
                      QUICK_DEPLOYMENT_STEPS.CLOUD_CONNECTION_STEP,
                    )
                  }
                >
                  {t("Create a Quick Deployment")}
                </Button>
              </div>
            )}
          </CardContent>
          {deploymentDialog && (
            <QuickDeploymentDialog
              connectionSummary={connectionSummary}
              environments={environments}
              cloudConnectionSelected={
                cloudConnectionCount === 1 ? selectedCloud : {}
              }
              environmentSelected={
                environments?.length === 1 ? environments[0] : {}
              }
              currentStep={currentStep}
              onClose={() => {
                setDeploymentDialog(false);
              }}
            />
          )}
        </Card>
      </Grid>
    </Grid>
  );
};

export default QuickDeployment;
