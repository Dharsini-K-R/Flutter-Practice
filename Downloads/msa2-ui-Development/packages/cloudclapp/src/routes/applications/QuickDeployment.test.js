import React from "react";
import { renderWithProvider } from "cloudclapp/src/utils/test-utils";
import QuickDeployment from "./QuickDeployment";
import initialState from "cloudclapp/src/store/initialState.js";

const mockState = {
  ...initialState,
};
describe("Quick Deployment component", () => {
  it("renders correctly", () => {
    const { container, getByText } = renderWithProvider({
      childComponent: <QuickDeployment />,
      mockState,
    });
    expect(container.firstChild).toMatchSnapshot();

    expect(getByText("Choose the application you want to deploy")).toBeTruthy();
    expect(getByText("Create a Quick Deployment")).toBeTruthy();
  });
});
