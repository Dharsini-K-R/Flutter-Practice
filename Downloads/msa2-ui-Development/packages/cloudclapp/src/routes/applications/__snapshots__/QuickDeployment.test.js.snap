// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quick Deployment component renders correctly 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-space-between"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
  >
    <p
      class="MuiTypography-root makeStyles-deploymentText MuiTypography-body1"
      id="choose_application_to_deploy"
    >
       
      Choose the application you want to deploy
       
    </p>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-deploymentButton"
      id="dockerhub-deployment-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Create a Quick Deployment
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
